scripts/shikify.ts

```diff
@@ -0,0 +1,68 @@
+import { createHighlighter, bundledLanguages, BundledTheme } from 'shiki';
+import * as fs from 'fs/promises';
+import * as path from 'path';
+import { parse } from 'node-html-parser';
+import { LangiumTextMateContent } from '../hugo/content/playground/constants.js';
+
+const mainPath = 'public/docs';
+const files = await fs.readdir(mainPath, {
+    recursive: true,
+    encoding: 'utf-8'
+});
+const htmlFiles = files.filter(file => file.endsWith('.html'));
+
+const darkTheme: BundledTheme = 'github-dark'
+const lightTheme: BundledTheme = 'github-light';
+
+const highlighter = await createHighlighter({
+    themes: [darkTheme, lightTheme],
+    langs: [LangiumTextMateContent as any, ...Object.keys(bundledLanguages)]
+});
+
+for (const file of htmlFiles) {
+    // Grab the file, and parse it into a DOM
+    const filePath = path.join(mainPath, file);
+    console.log(`Processing ${file}`);
+    const content = await fs.readFile(filePath, { encoding: "utf-8" });
+    const dom = parse(content);
+
+    // This isn't a particularly smart query implementation,
+    // so lets take the simple route and just grab all of the pre tags
+    const codeBlocks = dom.querySelectorAll("pre")
```

Comment:
Very pragmatic :D... but what about a more official way? Are there no `shiki` plugins or addons for Hugo?

I googled it and discovered your source for this code:
https://blog.puzzmo.com/posts/2024/06/23/shiki-hugo/
We should add a link in case of any improvements in their comments (Maybe we have some trouble with this code, then it is good to look up this link first).

WDYT?